.github/workflows/run-shared-tests.yml

```diff
@@ -53,6 +53,7 @@ on:
 jobs:
   run_e2e_tests:
     runs-on: ubuntu-latest
+    concurrency: inputs.run_e2e_tests
```

Comment:
This won't actually read the input value. In any case I guess we want something environment specific rather than a boolean?